packages/foundry-sdk-generator/src/generate/betaClient/minifyBundle.ts

```diff
@@ -150,16 +223,47 @@ export class ProjectMinifier {
               const importsFromNamespace = new Set(
                 exportSourceFile.getExportedDeclarations().keys(),
               );
-              this.pushNextVisit(exportSourceFile, importsFromNamespace);
+              this.pushNextVisit(
+                exportSourceFile,
+                importsFromNamespace,
+                "minifyProjectA",
+              );
             }
           }
         }
 
         const namedExports = declaration.getNamedExports();
 
+        function deleteFromSet(set: Set<string>, toDelete: Set<string>) {
+          for (const d of toDelete) {
+            set.delete(d);
+          }
+        }
+
         // export * from "..."
         if (namedExports.length === 0) {
-          this.pushNextVisit(exportSourceFile, visitImports.imports);
```

Comment:
This here meant that even if the current file already resolved `fooBar` from an earlier export, that `fooBar` would be marked as keeper in the `export *` which is including more than needed.